votor/src/common.rs

```diff
@@ -32,21 +32,32 @@ impl VoteType {
     }
 }
 
-/// Lookup from `CertificateId` to the `VoteType`s that contribute,
-/// as well as the stake fraction required for certificate completion.
+/// For a given [`CertificateType`], returns the fractional stake and the [`Vote`]s required to construct it.
 ///
-/// Must be in sync with `vote_to_certificate_ids`
-pub const fn certificate_limits_and_vote_types(
-    cert_type: &CertificateType,
-) -> (f64, &'static [VoteType]) {
```

Comment:
We originally wrote it this way just because we don't want to dynamically create and return a vector every time this is called.

What you are doing might be okay, but can you do some benchmarking to make sure we improved performance?

Reply:
Fair point.  I've now updated the function so that it no longer returns a `Vec` so even though it is not `const`, it should be returning the data on the stack and no actual memory allocation should be happening.  

This is also cleaner for the users of this function as they do not have to loops anymore which will make them more efficient.

Reply:
I would feel more comfortable landing this PR with a benchmark, because this sits on the critical path of inserting votes into the pool.

We can put the benchmark into a separate PR though.